internal/test/cleanup/cleanup.go

```diff
@@ -99,12 +99,20 @@ func CleanUpCloudstackTestResources(ctx context.Context, clusterName string, dry
 	if err != nil {
 		return fmt.Errorf("building cmk executable: %v", err)
 	}
-	cmk := executableBuilder.BuildCmkExecutable(tmpWriter, *execConfig)
+	cmk := executableBuilder.BuildCmkExecutable(tmpWriter, execConfig.Profiles)
 	defer cmk.Close(ctx)
 
-	if err := cmk.ValidateCloudStackConnection(ctx); err != nil {
-		return fmt.Errorf("validating cloudstack connection with cloudmonkey: %v", err)
+	failedProfiles := []string{}
+	errors := []error{}
+	for _, profile := range execConfig.Profiles {
+		if err := cmk.CleanupVms(ctx, profile.Name, clusterName, dryRun); err != nil {
+			failedProfiles = append(failedProfiles, profile.Name)
+			errors = append(errors, err)
+		}
 	}
 
-	return cmk.CleanupVms(ctx, clusterName, dryRun)
+	if len(failedProfiles) > 0 {
+		return fmt.Errorf("cleaning up VMs: profiles=%+v, errors=%+v", failedProfiles, errors)
```

Comment:
Should we be doing a profile=error mapping instead if that is the intention here?